File: rust/soma10numeros/soma10numeros.rs


fn main() {

    let lista = 0..11;
    let mut total = 0;
    for i in lista {
        total = total + i;
    }

    println!("Na mão");
    println!("{}", total);

    println!("Automatico");
    let total = (0..11).fold(0, |soma, item_atual| soma + item_atual);
    println!("{}", total);
}

File: c#/soma10numeros/soma10numeros.cs
using System.Collections;
using System;
using System.Linq;

var lista = Enumerable.Range(1, 10);

var total = 0;
foreach (var item in lista)
{
    total += item;
}

Console.WriteLine($"Na mão {total}");

Console.WriteLine($"Automatico {lista.Sum()}");